designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+
+**4. Both legacy path and consolidated path exist, but legacy path is not a symbolic link.**
+
+**4.1 If NuGet could determine how to consolidate the contents of the two**
+If NuGet could determine how to consolidate the contents of the two, NuGet will consolidate the two into consolidated path, and set up the legacy path to be a symbolic link to the consolidated path. After consolidation, NuGet will get config file from consolidated path.
+No warning will be shown.
+
+**4.2 If NuGet could not determine how to consolidate the contents of the two**
+If NuGet could not determine how to consolidate the contents of the two, NuGet will get user wide NuGet.Config file from **legacy path**.
+NuGet will show a warning, encouraging customers to consolidate the contents of the two and only keep the one in consolidated path. If customers finish that, the next time, it will jump to scenario 3. legacy path does not exist, but consolidated path exists, so NuGet will help them set up the legacy path to be a symbolic link to the consolidated path. 
+
+Warning: there are two user-wide NuGet.Config files. Put the wanted config file in consolidated path, delete the one from legacy path. Next time, NuGet will set up the legacy path to be a symbolic link to the consolidated path. For more details, please refer to: a link of document.
+
+#### Pros: 
+If customer uses gloabal.json to pin different versions of dotnet dynamically, the old version of dotnet will pick config file from legacy path, which is actually the same file with the one in consolidated path. So all versions of dotnet could maintain a single copy of NuGet.Config files. 
+
+#### Cons: 
+Legacy path still exists for all scenarios, even if the NuGet.Config file is only in consolidated path, legacy path will be created as a symbolic link.
+
+If customer sets legacy path wrongly by themselves(not following instructions in warning of scenario 4.2 and let NuGet to do that for them), e.g. a symbolic link pointing to a wrong path, NuGet is able to detect that, but NuGet won't know if it's on purpose or it's a wrong setting. So NuGet won't break customers in this situation.
+
+### Scenarios from the developer side
+**1. Visual Studio for Mac** 
+
+  It's in .NET Framework code path so it will not be affected by this change.
+
+**2. Mono CLI (NuGet)**
+
+  It's in .NET Framework code path so it will not be affected by this change.
+
+**3. dotnet CLI (NuGet)**
+
+  On windows, it will not be affected by this change.
+  On Linux/Mac, this change will be applied.
+
+**4. CI/CD (Headless CLI)**
+
+  On windows, it will not be affected by this change.
+  On Linux/Mac, this change will be applied. But considering most users will put the URLs in a NuGet.Config file in the repository, so the user wide NuGet.Config in both legacy path and consolidated path should be the default content in most cases. Not sure if there is any data we could use to prove this assumption.
+
+
+### Challenges
```

Comment:
This section still appears to have a lot of open questions, which makes me feel that the spec is not ready to be merged.